contract/tests/test_basics.rs

```diff
@@ -156,6 +165,216 @@ async fn test_view_call_verify_transaction_inclusion() -> Result<(), Box<dyn std
     Ok(())
 }
 
+fn read_blocks_from_json(path: &str) -> Vec<Vec<Header>> {
+    let file = File::open(path).expect("Unable to open file");
+    let reader = BufReader::new(file);
+    serde_json::from_reader(reader).unwrap()
+}
+
+#[tokio::test]
+async fn test_submit_blocks_for_period() -> Result<(), Box<dyn std::error::Error>> {
+    let (contract, user_account, block_headers) = init_contract_from_file(2017).await?;
+
+    for block_headers_batch in &block_headers[1..] {
+        let outcome = user_account
+            .call(contract.id(), "submit_blocks")
+            .args_borsh(block_headers_batch.clone())
+            .deposit(STORAGE_DEPOSIT_PER_BLOCK)
+            .max_gas()
+            .transact()
+            .await?;
+
+        assert!(outcome.is_success());
+    }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_get_last_n_blocks() -> Result<(), Box<dyn std::error::Error>> {
+    let (contract, user_account, block_headers) = init_contract_from_file(2017).await?;
+
+    for block_headers_batch in &block_headers[1..=2] {
+        let outcome = user_account
+            .call(contract.id(), "submit_blocks")
+            .args_borsh(block_headers_batch.clone())
+            .deposit(STORAGE_DEPOSIT_PER_BLOCK)
+            .max_gas()
+            .transact()
+            .await?;
+
+        assert!(outcome.is_success());
+    }
+
+    let user_message_outcome = contract
+        .view("get_last_n_blocks_hashes")
+        .args_json(json!({"skip": 0, "limit": 0}))
+        .await?;
+
+    assert_eq!(user_message_outcome.json::<Vec<H256>>()?, vec![]);
+
+    let user_message_outcome = contract
+        .view("get_last_n_blocks_hashes")
+        .args_json(json!({"skip": 0, "limit": 200}))
+        .await?;
+
+    assert_eq!(user_message_outcome.json::<Vec<H256>>()?.len(), 97);
+
+    let user_message_outcome = contract
+        .view("get_last_n_blocks_hashes")
+        .args_json(json!({"skip": 200, "limit": 200}))
+        .await?;
+
+    assert_eq!(user_message_outcome.json::<Vec<H256>>()?, vec![]);
+
+    let user_message_outcome = contract
+        .view("get_last_n_blocks_hashes")
+        .args_json(json!({"skip": 10, "limit": 10}))
+        .await?;
+
+    let last_blocks = user_message_outcome.json::<Vec<H256>>()?;
+    assert_eq!(
+        last_blocks[0].to_string(),
+        "0000000000000000000aab4a401ac27b945057be99db4ccc9631da4bb0b9d746"
+    );
+
+    assert_eq!(
+        last_blocks[9].to_string(),
+        "0000000000000000000758a734884015e791dee8aced3dcce049753dc5aeeacb"
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_gc() -> Result<(), Box<dyn std::error::Error>> {
+    let (contract, user_account, block_headers) = init_contract_from_file(10).await?;
+
+    for block_headers_batch in &block_headers[1..=2] {
+        let outcome = user_account
```

Comment:
It is not clear how many blocks were submitted

Reply:
fixed: https://github.com/Near-One/btc-light-client-contract/pull/74/commits/9f535a870ccc7a22e6675d853455da3eb84574d2